var/spack/repos/builtin/packages/namd/package.py

```diff
@@ -0,0 +1,155 @@
+        shutil.copy('arch/Linux-x86_64.{0}'.format(lib),
+                    config_filename)
+        if lib == 'tcl':
+            filter_file(r'-ltcl8\.5',
+                        '-ltcl{0}'.format(self.spec['tcl'].version.up_to(2)),
+                        config_filename)
+
+    def _append_option(self, opts, lib):
+        if lib != 'python':
+            self._copy_arch_file(lib)
+        spec = self.spec
+        opts.extend([
+            '--with-{0}'.format(lib),
+            '--{0}-prefix'.format(lib), spec[lib].prefix
+        ])
+
+    @property
+    def arch(self):
+        plat = sys.platform
+        if plat.startswith("linux"):
+            plat = "linux"
+        march = platform.machine()
+        return '{0}-{1}'.format(plat, march)
+
+    @property
+    def build_directory(self):
+        return '{0}-spack'.format(self.arch)
+
+    def edit(self, spec, prefix):
+        with working_dir('arch'):
+            with open('{0}.arch'.format(self.build_directory), 'w') as fh:
+                # this options are take from the default provided
+                # configuration files
+                optims_opts = {
+                    'gcc': '-m64 -O3 -fexpensive-optimizations -ffast-math',
+                    'intel': '-O2 -ip'
+                }
+
+                optim_opts = optims_opts[self.compiler.name] \
+                    if self.compiler.name in optims_opts else ''
+
+                fh.write('\n'.join([
+                    'NAMD_ARCH = {0}'.format(self.arch),
+                    'CHARMARCH = ',
+                    'CXX = {0.cxx} {0.cxx11_flag}'.format(
+                        self.compiler),
+                    'CXXOPTS = {0}'.format(optim_opts),
+                    'CC = {0}'.format(self.compiler.cc),
+                    'COPTS = {0}'.format(optim_opts),
+                    ''
+                ]))
+
+        self._copy_arch_file('base')
+
+        opts = ['--charm-base', spec['charm'].prefix]
+        fftw_version = spec.variants['fftw'].value
+        if fftw_version == 'none':
+            opts.append('--without-fftw')
+        elif fftw_version == 'mkl':
+            self._append_option(opts, 'mkl')
+        else:
+            _fftw = 'fftw{0}'.format('' if fftw_version == '2' else '3')
+
+            self._copy_arch_file(_fftw)
+            opts.extend(['--with-{0}'.format(_fftw),
+                         '--fftw-prefix', spec['fftw'].prefix])
+
+        interface_type = spec.variants['interface'].value
+        if interface_type != 'none':
+            self._append_option(opts, 'tcl')
+
+            if interface_type == 'python':
+                self._append_option(opts, 'python')
+        else:
+            opts.extend([
+                '--without-tcl',
+                '--without-python'
+            ])
+
+        config = Executable('./config')
+
+        config(self.build_directory, *opts)
+
+    def build(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make()
```

Comment:
I don't think you need to define `build` anymore as this is the default implementation.